### PR TITLE
Avoids linking nghttp2 in Carthage project

### PR DESCRIPTION
**Pull request by WilliamIzzo83**
This should address #449

### CODE REVIEW

File: patch-carthage-project.rb [hunk 1]
```diff
@@ -8,9 +8,23 @@
 targets_to_remove.each do |target|
   target.remove_from_project
 end
+
+# 2) Prevent linking of nghttp2 library
+carthage_nghttp2_unlink_targets = [ "BoringSSL", "CgRPC", "SwiftGRPC" ]
+targets_to_unlink = project.targets.select { |target| carthage_nghttp2_unlink_targets.include?(target.name) }
+ldflags_to_remove = [ "-L/usr/local/Cellar/nghttp2/1.38.0/lib", "-lnghttp2" ]
```
Comment (MrMage) [inline patch-carthage-project.rb]: Would it be possible to make this more generic, removing any `LDFLAG` containing `nghttp2`?

File: SwiftGRPC-Carthage.xcodeproj/GeneratedModuleMap/CNIONghttp2/module.modulemap [hunk 2]
```diff
@@ -1,4 +1,4 @@
 module CNIONghttp2 {
-    umbrella "/Users/mrebello/Development/grpc-swift/.build/checkouts/swift-nio-http2.git-1684232237084789971/Sources/CNIONghttp2/include"
```
Comment (MrMage) [inline SwiftGRPC-Carthage.xcodeproj/GeneratedModuleMap/CNIONghttp2/module.modulemap]: I'm a bit surprised by all the `.git-1684232237084789971` suffixes disappearing. I'd expect the values to have changed, but that there would still be a suffix present. Which Swift version did you use to generate the new project? (I think we should stick with Swift 4.2 for now, to ensure that we don't break the build for Swift 4.2 users by generating the project with Swift 5.) See https://medium.com/xcblog/switching-swift-versions-inside-xcode-using-toolchains-755b28831c43 on selecting the Swift 4.2 toolchain.

File: patch-carthage-project.rb (hunk 1 above)
Comment (WilliamIzzo83) [reply patch-carthage-project.rb]: I can scan every target in the scheme if that is all right with you

File: SwiftGRPC-Carthage.xcodeproj/GeneratedModuleMap/CNIONghttp2/module.modulemap (hunk 2 above)
Comment (WilliamIzzo83) [reply SwiftGRPC-Carthage.xcodeproj/GeneratedModuleMap/CNIONghttp2/module.modulemap]: Of course: I'm using swift 5. I'll fix that

File: patch-carthage-project.rb (hunk 1 above)
Comment (MrMage) [reply patch-carthage-project.rb]: Sounds good, but I was more thinking of removing any flag that contains `nghttp2`, even if there is not an exact match.

Comment (WilliamIzzo83) [reply patch-carthage-project.rb]: Done

File: SwiftGRPC-Carthage.xcodeproj/GeneratedModuleMap/CNIONghttp2/module.modulemap (hunk 2 above)
Comment (WilliamIzzo83) [reply SwiftGRPC-Carthage.xcodeproj/GeneratedModuleMap/CNIONghttp2/module.modulemap]: Done

**Pull request merged**